app/src/main/java/com/keylesspalace/tusky/adapter/ThreadAdapter.kt

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2017 Andrew Dawson
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+package com.keylesspalace.tusky.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.interfaces.StatusActionListener
+import com.keylesspalace.tusky.util.StatusDisplayOptions
+import com.keylesspalace.tusky.viewdata.StatusViewData
+
+class ThreadAdapter(
+    private val statusDisplayOptions: StatusDisplayOptions,
+    private val statusActionListener: StatusActionListener
+) : RecyclerView.Adapter<StatusBaseViewHolder>() {
+    private val statuses = mutableListOf<StatusViewData.Concrete>()
+    var detailedStatusPosition: Int = RecyclerView.NO_POSITION
+        private set
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StatusBaseViewHolder {
+        return when (viewType) {
+            VIEW_TYPE_STATUS -> {
+                val view = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_status, parent, false)
+                StatusViewHolder(view)
+            }
+            VIEW_TYPE_STATUS_DETAILED -> {
+                val view = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_status_detailed, parent, false)
+                StatusDetailedViewHolder(view)
+            }
+            else -> {
+                val view = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_status, parent, false)
+                StatusViewHolder(view)
+            }
```

Comment:
I'd either put this in a if/else to not have 3 branches or throw an exception in the third branch (also in the other adapters)